enabling cache dir as a secondary FS mount

cpoole opened the pull request:
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md

<!--
Thank you for contributing to jfrog/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a TravisCI
will run across your changes, do linting and then install the chart.
Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

**What this PR does / why we need it**:
If using the filesystem backend on an NFS mount it is nice to have a second volume mounted for the cache directory since NFS is far too slow for cache. 

This enables an explicit NFS type check in the filestore binarystoreXML and creates a volume for cacheFs


**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #


**Special notes for your reviewer**:

eldada commented on stable/artifactory/README.md:
````diff
@@ -119,6 +119,16 @@ To use an NFS server as your cluster's storage, you need to
 ...
 ```
 
+To enable a second volume for cache storage: 
````
Can you give some more context to what problem does this feature solve?

eldada commented on stable/artifactory/values.yaml:
```diff
@@ -288,6 +289,12 @@ artifactory:
 
     maxCacheSize: 50000000000
     cacheProviderDir: cache
+    cacheFs:
```
Missing `cacheFs.persistence.storageClass`

eldada commented on stable/artifactory/README.md:
```diff
@@ -618,7 +628,10 @@ The following table lists the configurable parameters of the artifactory chart a
+| `artifactory.persistence.cacheFs.enabled` | Enabled a cache persistent volume | `false` |
+| `artifactory.persistence.cacheFs.persistence.accessMode` | Access mode for the cache persistent volume | `ReadWriteOnce`         |
+| `artifactory.persistence.cacheFs.persistence.bytes`      | Desired size in bytes of the persistent volume | `182536110080`       |
+| `artifactory.persistence.cacheFs.persistence.mountPath`  | Location on the filesystem to mount the cache volume | `/jfrog/cache` |
```
Missing `artifactory.persistence.cacheFs.persistence.storageClass`

eldada commented on stable/artifactory/README.md:
```diff
@@ -618,7 +628,10 @@ The following table lists the configurable parameters of the artifactory chart a
-| `artifactory.persistence.cacheProviderDir`  | the root folder of binaries for the filestore cache. If the value specified starts with a forward slash ("/") it is considered the fully qualified path to the filestore folder. Otherwise, it is considered relative to the *baseDataDir*. | `cache`                   |
+| `artifactory.persistence.cacheFs.enabled` | Enabled a cache persistent volume | `false` |
+| `artifactory.persistence.cacheFs.persistence.accessMode` | Access mode for the cache persistent volume | `ReadWriteOnce`         |
+| `artifactory.persistence.cacheFs.persistence.bytes`      | Desired size in bytes of the persistent volume | `182536110080`       |
```
Default value you set is `60000000000`

eldada commented:
@cpoole - can you review my comments?

cpoole commented:
hey @eldada I addressed your comments. There isn't currently documentation in the readme for the default behavior of the storageclasses so I just put `see yalues.yaml` and put the explanation in values.yaml. 

That seems to be what the chart does in other places as well. 

As for adding to artifactory-ha: I wont have any time this week to look into it and I do not have a license for artifactory HA. Are the automated tests sufficient enough to actually validate the cache directory changes?

danielezer commented on stable/artifactory/README.md:
```diff
@@ -618,7 +630,11 @@ The following table lists the configurable parameters of the artifactory chart a
 | `artifactory.persistence.binarystoreXml` | Artifactory binarystore.xml template             | See `values.yaml`                    |
 | `artifactory.persistence.customBinarystoreXmlSecret` | A custom Secret for binarystore.xml  | ``                                   |
 | `artifactory.persistence.maxCacheSize`   | The maximum storage allocated for the cache in bytes. | `50000000000`                   |
-| `artifactory.persistence.cacheProviderDir`  | the root folder of binaries for the filestore cache. If the value specified starts with a forward slash ("/") it is considered the fully qualified path to the filestore folder. Otherwise, it is considered relative to the *baseDataDir*. | `cache`                   |
```
Why did you remove this? I see it is still used.

danielezer commented:
This was added in PR #563
